pkg/asset/imagebased/configimage/clusterconfiguration.go

```diff
@@ -0,0 +1,174 @@
+package configimage
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	lca "github.com/openshift-kni/lifecycle-agent/api/seedreconfig"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/agent"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/password"
+	"github.com/openshift/installer/pkg/asset/tls"
+)
+
+var (
+	clusterConfigurationFilename = filepath.Join(clusterConfigurationDir, "manifest.json")
+
+	_ asset.WritableAsset = (*ClusterConfiguration)(nil)
+)
+
+// ClusterConfiguration generates the image-based installer cluster configuration asset.
+type ClusterConfiguration struct {
+	File   *asset.File
+	Config *lca.SeedReconfiguration
+}
+
+// Name returns a human friendly name for the asset.
+func (*ClusterConfiguration) Name() string {
+	return "Image-based installer cluster configuration"
+}
+
+// Dependencies returns all of the dependencies directly needed to generate
+// the asset.
+func (*ClusterConfiguration) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.ClusterID{},
+		&tls.KubeAPIServerLBSignerCertKey{},
+		&tls.KubeAPIServerLocalhostSignerCertKey{},
+		&tls.KubeAPIServerServiceNetworkSignerCertKey{},
+		&tls.AdminKubeConfigSignerCertKey{},
+		&IngressOperatorSignerCertKey{},
+		&password.KubeadminPassword{},
+		&agent.OptionalInstallConfig{},
```

Comment:
I think we might need a specific one for ibi and validate that the install config is compatible with what ibi (currently) supports (none platform sno, something like [this](https://github.com/mresvanis/installer/blob/4cf21ee040982fff267cda2821a6fab87bd4f30e/pkg/asset/agent/installconfig.go#L211))

Reply:
Definitely, that's a great point and one that I wanted to refactor as well! I'll get on with it and update this PR.